### ISSUE TITLE
Clapper Flatpak / theming in Ubuntu 24.04

### ISSUE THREAD

**Issue by Coeur-Noir** [Rafostar/clapper#445]
Hi,

not a show-stopper, just a matter of taste and visual consistency.

Using Clapper installed through Flatpak on Ubuntu 24.04 ( since the .deb version in their repository is older. )

Also installed all Yaru themes through Flatpak.

I expected therefore to see Clapper themed accordingly ( ie. orange for accentuation color instead of blue ).
Nope. Still blue.

Is there a way to achieve a Yaru theming on Clapper ? The ( older ) .deb package is Yaru themed though ( but has too many issues. )

**Comment by Rafostar** [comment]
Flatpak Yaru theme is only for GTK3 apps AFAIK. Probably best hack to get this working with GTK4 is:

Copy themes from `/usr/share/themes` to `.themes` folder in your Home directory. If it down not exists, create it. Then apply permissions:

```sh
sudo flatpak override --filesystem=$HOME/.themes com.github.rafostar.Clapper
sudo flatpak override --env=GTK_THEME=Yaru-dark com.github.rafostar.Clapper
```

**Comment by Coeur-Noir** [comment]
Oh thanks !

Here is what I've tried, with FlatSeal : 
![Flatseal_Clapper](https://github.com/Rafostar/clapper/assets/1895947/fde31b92-9c31-4cfe-b1c5-eba0df55dc6f)

and the quite interesting result : 
![Clapper_file-chooser](https://github.com/Rafostar/clapper/assets/1895947/94ca2d05-1f16-478e-9347-dddaee9571df)

and then I went back to « normal » : 
![Clapper_initial_situation](https://github.com/Rafostar/clapper/assets/1895947/8103d939-9fe1-42dd-a705-86d571ca1bc1)

What bugs me a little is the file-chooser which is correctly themed with both settings.

Anyway it's not a functional issue and maybe not really a bug of yours.
Maybe related to ( fresh ) Ubuntu 24.04, Yaru-theme or some FlatPak subtleties ( …my first steps with FlatPak also. )

**Comment by Rafostar** [comment]
Clapper uses your native system file chooser via portal, so it's not surprising.

**Comment by Coeur-Noir** [comment]
Yes, it's the native one but it's also GTK4 and correctly themed.

So there might be a way...
-- 
Envoyé depuis Motorola g84 via le courrieleur libre K-9 Mail ( prochainement Thunderbird ).

Le 3 mai 2024 08:30:56 GMT+02:00, "Rafał Dzięgiel" ***@***.***> a écrit :

**Comment by Rafostar** [comment]
If Yaru tries to theme a libadwaita app, then this will not be supported (cause libadwaita does not support theming). The only thing theme should be doing for an app is to apply custom accent colors.

First, check if you have `$HOME/.config/gtk-4.0/gtk.css` with predefined theme colors.
If you do not, then:

```sh
mkdir -p $HOME/.config/gtk-4.0
echo -e '@define-color accent_color #E95420;\n@define-color accent_bg_color #E95420;' > $HOME/.config/gtk-4.0/gtk.css
```

Finally add permission for Clapper to read that directory:
```sh
sudo flatpak override --filesystem=xdg-config/gtk-4.0:ro com.github.rafostar.Clapper
```

Result:

![image](https://github.com/Rafostar/clapper/assets/40623528/1e0f4769-5172-4c5f-a676-d15e52d75882)
